### PR TITLE
Update spotify.snap to 1.2.22.982.g794acc0a (+ some fixes)

### PR DESCRIPTION
**Pull request by Erick555**
Closes https://github.com/flathub/com.spotify.Client/pull/259

Closes https://github.com/flathub/com.spotify.Client/pull/260

Closes https://github.com/flathub/com.spotify.Client/issues/257

### CODE REVIEW

File: com.spotify.Client.json [hunk 1]
```diff
@@ -17,12 +19,17 @@
...
         "--own-name=org.mpris.MediaPlayer2.spotify",
         "--filesystem=xdg-music:ro",
         "--filesystem=xdg-pictures:ro",
         "--env=LD_LIBRARY_PATH=/app/lib",
         "--env=TMPDIR=/tmp"
     ],
+    "build-options": {
+        "append-path": "/usr/lib/sdk/vala/bin",
+        "prepend-ld-library-path": "/usr/lib/sdk/vala/lib"
+    },
```
Comment (TingPing) [inline com.spotify.Client.json:32]: Why are you including Vala? Surely every dep supports disabling it?

Comment (TingPing) [reply com.spotify.Client.json:32]: I ended up merging this with Vala removed.

Comment (Erick555) [reply com.spotify.Client.json:32]: In my testing `ayatana-ido` failed to build without vala. If it works without it then it's great.

Comment (TingPing) [reply com.spotify.Client.json:32]: I had to patch it :disappointed:

Comment (bbhtt) [reply com.spotify.Client.json:32]: I just added it to Liferea on GNOME 45 without vala and it seems to work fine? No patching needed. https://github.com/flathub/net.sourceforge.liferea/commit/071f44ed8cc8bb0813177559d86efac1141ac14d

Comment (bbhtt) [reply com.spotify.Client.json:32]: Ah nvm, forgot that one comes with vala

**Pull request merged**